jina/serve/executors/__init__.py

```diff
@@ -124,7 +122,12 @@ def _add_runtime_args(self, _runtime_args: Optional[Dict]):
             self.runtime_args = SimpleNamespace()
 
     def _init_monitoring(self):
-        if hasattr(self.runtime_args, 'metrics_registry'):
+        if not hasattr(self.runtime_args, 'metrics_registry'):
```

Comment:
this should not be done here. This should be handled in the setting of these `arguments`. `_init_monitoring` should not alter the `runtime_args`